### PR TITLE
Logging: support request-correlated logging in App Engine standard python37 runtime

### PR DESCRIPTION
**Pull request by myelin**
The python37 runtime provides the GOOGLE_CLOUD_PROJECT environment variable instead of GCLOUD_PROJECT, and for log messages to be correlated with the request in StackDriver, their 'trace' value has to be passed as a top level member of the log record rather than as an appengine.googleapis.com/trace_id label.

### CODE REVIEW

File: logging/google/cloud/logging/handlers/app_engine.py [hunk 1]
```diff
@@ -65,9 +68,9 @@ def get_gae_resource(self):
         gae_resource = Resource(
             type='gae_app',
             labels={
-                'project_id': os.environ.get(_GAE_PROJECT_ENV),
-                'module_id': os.environ.get(_GAE_SERVICE_ENV),
-                'version_id': os.environ.get(_GAE_VERSION_ENV),
+                'project_id': self.project_id,
+                'module_id': os.environ.get(_GAE_SERVICE_ENV, ''),
```
Comment (theacodes) [inline logging/google/cloud/logging/handlers/app_engine.py]: Do you mind going ahead and promoting `module_id` and `version_id` to public properties?

Comment (myelin) [reply logging/google/cloud/logging/handlers/app_engine.py]: Done :)

**Comment by myelin** [comment]
I broke something with this last CL -- fix coming soon.

**Comment by theacodes** [comment]
Thank you, @myelin!

File: logging/google/cloud/logging/handlers/app_engine.py [hunk 2]
```diff
@@ -54,6 +55,11 @@ def __init__(self, client,
         self.name = name
         self.client = client
         self.transport = transport(client, name)
+        self.project_id = os.environ.get(
+            _GAE_PROJECT_ENV_FLEX,
+            os.environ.get(_GAE_PROJECT_ENV_STANDARD, ''))
```
Comment (tseaver) [inline logging/google/cloud/logging/handlers/app_engine.py]: This spelling hides a branch (see comment below about another test needed).

File: logging/tests/unit/handlers/test_app_engine.py [hunk 3]
```diff
@@ -30,15 +30,18 @@ def _make_one(self, *args, **kw):
...
         client = mock.Mock(project=self.PROJECT, spec=['project'])
 
-        with mock.patch('os.environ', new={_GAE_PROJECT_ENV: 'test_project',
-                                           _GAE_SERVICE_ENV: 'test_service',
-                                           _GAE_VERSION_ENV: 'test_version'}):
+        with mock.patch('os.environ', new={
+            _GAE_PROJECT_ENV_STANDARD: 'test_project',
+            _GAE_SERVICE_ENV: 'test_service',
+            _GAE_VERSION_ENV: 'test_version',
+        }):
```
Comment (tseaver) [inline logging/tests/unit/handlers/test_app_engine.py]: We really should test for the case that the flex variable is set, and that it overrides the standard one.

Comment (myelin) [reply logging/tests/unit/handlers/test_app_engine.py]: Done in #6138.

**Pull request merged**